js/main.js

```diff
@@ -0,0 +1,312 @@
+  var weaponsPage = document.querySelector('.weapons');
+  var searchResults = document.querySelector('.search-results');
+
+  if (viewName === 'home') {
+    homePage.className = 'home';
+    agentsPage.className = 'agents hidden';
+    mapsPage.className = 'maps hidden';
+    weaponsPage.className = 'weapons hidden';
+    searchResults.className = 'search-results hidden';
+  } else if (viewName === 'agents') {
+    homePage.className = 'home hidden';
+    agentsPage.className = 'agents';
+    mapsPage.className = 'maps hidden';
+    weaponsPage.className = 'weapons hidden';
+    searchResults.className = 'search-results hidden';
+  } else if (viewName === 'maps') {
+    homePage.className = 'home hidden';
+    agentsPage.className = 'agents hidden';
+    mapsPage.className = 'maps';
+    weaponsPage.className = 'weapons hidden';
+    searchResults.className = 'search-results hidden';
+  } else if (viewName === 'weapons') {
+    homePage.className = 'home hidden';
+    agentsPage.className = 'agents hidden';
+    mapsPage.className = 'maps hidden';
+    weaponsPage.className = 'weapons';
+    searchResults.className = 'search-results hidden';
+  } else if (viewName === 'search-results') {
+    homePage.className = 'home hidden';
+    agentsPage.className = 'agents hidden';
+    mapsPage.className = 'maps hidden';
+    weaponsPage.className = 'weapons hidden';
+    searchResults.className = 'search-results';
+
+  }
+}
+
+// event listener for tabs and drop down
+var $homeTab = document.querySelector('.home-tab');
+var $agentsTab = document.querySelector('.agents-tab');
+var $mapsTab = document.querySelector('.maps-tab');
+var $weaponsTab = document.querySelector('.weapons-tab');
+var $searchButton = document.querySelector('.search-button');
+
+$homeTab.addEventListener('click', function () {
+  viewSwap('home');
+  $dropdownModal.className = 'dropdown-menu hidden';
+
+});
+
+$agentsTab.addEventListener('click', function () {
+  viewSwap('agents');
+});
+
+$mapsTab.addEventListener('click', function () {
+  viewSwap('maps');
+});
+
+$weaponsTab.addEventListener('click', function () {
+  viewSwap('weapons');
+});
+
+$searchButton.addEventListener('click', function () {
+  viewSwap('search-results');
+});
+
+var $agentsDrop = document.querySelector('.agents-drop');
+var $mapsDrop = document.querySelector('.maps-drop');
+var $weaponsDrop = document.querySelector('.weapons-drop');
+
+$agentsDrop.addEventListener('click', function () {
+  viewSwap('agents');
+  $dropdownModal.className = 'dropdown-menu hidden';
+});
+
+$mapsDrop.addEventListener('click', function () {
+  viewSwap('maps');
+  $dropdownModal.className = 'dropdown-menu hidden';
+
+});
+
+$weaponsDrop.addEventListener('click', function () {
+  viewSwap('weapons');
+  $dropdownModal.className = 'dropdown-menu hidden';
+
+});
+
+var xhr = new XMLHttpRequest();
+xhr.onreadystatechange = function () {
+  if (this.readyState === 4 && this.status === 200) {
```

Comment:
Instead of using `onreadystatechange`, I believe you should just use the `load` event. This would make sure that the following function would only run when content has been retrieved from the api